### ISSUE TITLE
Improve code coverage for System.Xml.XPath

### ISSUE THREAD

**Issue by sepidehMS** [dotnet/corefx#16646]
- [ ]  System.Xml.XPath.Extensions 20%
- [ ]  System.Xml.XPath.
- [ ]  System.Xml.XPath.
- [ ]  System.Xml.XPath.

**Comment by danmosemsft** [comment]
Info on code coverage work:

1) Here's the all up report
https://ci.dot.net/job/dotnet_corefx/job/master/job/code_coverage_windows/Code_Coverage_Report/
Caveat - doesn't show coverage of any types whose implementation is in corelib (eg String, etc). If you want to look at that for now you must gather manually using instructions below.

2) Docs on code coverage work are here
https://github.com/dotnet/corefx/blob/master/Documentation/building/code-coverage.md
and some in
https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md

3) There's innumerable places to add coverage -- we would love to focus on the most heavily used types and the least covered types to get the most benefit.

**Comment by WinCPP** [comment]
@danmosemsft @sepidehMS I think I will start with the 2nd one  System.Xml.XPath.XPathNavigator (21%) coverage. Just in case anyone else also wants to work in parallel, then he / she can pick from the other two.

Looks like the files are huge and lots of it to do. So would it be ok if I submit coverage incrementally in different PRs. Also just in case I get exhausted after sufficient submission, is it ok if I switch to some other coding / bug fix issue and then come back here? Please assign this issue to me for now.

Thanks!!

**Comment by danmosemsft** [comment]
Yes, it's fine to submit incrementally. Yes, any amount of work you do here is fine -- you're not signing up to fix everything. It's all appreciated.

**Comment by danmosemsft** [comment]
@sepidehMS @krwq did we already take a look at reusing Mono tests? If not, @WinCPP, we can do that if it makes more sense than writing new ones.

We can copy any tests from Mono into our tree, as long as we do this:
```
	1. Keep the existing copyright headers in place
		○ Note: If you are porting just portions of the file over, you still need to bring over all the existing copyright headers.
	2. Add the following copyright headers – NOTE this is missing the middle line of the ‘regular’ CoreFx copyright headers:
// Licensed to the .NET Foundation under one or more agreements.
// See the LICENSE file in the project root for more information.
```

**Comment by sepidehMS** [comment]
@danmosemsft No, we haven't looked at Mono tests for Xml yet.

**Comment by danmosemsft** [comment]
OK @WinCPP I think that's a great place I recommend you start, we can increase code coverage very quickly that way and also discover any functional inconsistencies. Take a look in the MOno repo...

**Comment by WinCPP** [comment]
@danmosemsft sure!

**Comment by WinCPP** [comment]
@danmosemsft @sepidehMS I am trying to understand mono repo structure to locate the relevant tests. I see a folder hierarchy that has 'CoreFx' in its path. `E.g. M:\mono\external\corefx\src\System.Private.Xml\tests`. Is that the place to look for the tests. Could you please give quick hint for where to look into mono repo, for any one of the 3 in the [first comment](https://github.com/dotnet/corefx/issues/16646#issue-211564210), it will help me get started sooner... Thanks!

**Comment by danmosemsft** [comment]
@krwq also

**Comment by danmosemsft** [comment]
I think they're here
https://github.com/mono/mono/blob/master/mcs/class/System.XML/Test/System.Xml.XPath

**Comment by WinCPP** [comment]
@danmosemsft , sorry catching up this thread... got busy with some urgent personal work... I'll start with System.Xml.XPath.XPathNavigator coverage...

**Comment by WinCPP** [comment]
@danmosemsft @sepidehMS about the copyright header... the mono file has this header...

```C#
//
// <FQ Class Name>
//
// Authors:
//   Auth 1 <email ID>
//   Auth 2 <email ID>
//   Auth 3 <email ID>
//
// (C) <Year> <Auth 1>
// (C) <Year> <Auth 2>
// (C) <Year Range> <Some Company Name>
//
```

I believe we are referring to just the '(C)` lines to be copied over, appended by lines mentioned in [this](https://github.com/dotnet/corefx/issues/16646#issuecomment-289871041) comment. Essentially the cumulative block should look like this...?
```C#
// (C) <Year> <Auth 1>
// (C) <Year> <Auth 2>
// (C) <Year Range> <Some Company Name>
// Licensed to the .NET Foundation under one or more agreements.
// See the LICENSE file in the project root for more information.
```

Please confirm. Thanks!

**Comment by danmosemsft** [comment]
@WinCPP that looks good, I'm pretty ocnfident those (C) lines are rather useless but it's easier to do this than for me to ask legal folk!

**Comment by danmosemsft** [comment]
Second thoughts, I'll ask, hold on

**Comment by WinCPP** [comment]
Hi @danmosemsft ... bumping the thread... Any inputs after your last comment in this thread ([here](https://github.com/dotnet/corefx/issues/16646#issuecomment-290539524)).

Is this something that I can still look into...?

**Comment by danmosemsft** [comment]
@wincpp just remove those bogus lines and use our regular license.

**Comment by WinCPP** [comment]
@sepidehkh @danmosemsft I am working on importing XPathNavigator tests from mono to CoreFx repo as discussed in the above thread. I have used `.\src\System.Private.Xml\tests\XPath\XPathDocument` as template to add new project for `XPathNavigatorTests`. I am facing build issues with the tests...

When I change to the directory `src\System.Private.Xml\tests\XPath\XPathDocument`, say for example, and issue `msbuild /t:RebuildAndTest` command, the tests fail with error that the required test dll is not found. This is happening for all the other existing test projects as well. As I see, this is a bit non-standard test project folder hierarchy. Usually the `msbuild /t:RebuildAndTest` command is issued in the relevant `.\Tests` folder in this case the required test case `.csproj` files are two folders down. Is that the issue? How do we run individual test case projects in this case? Appreciate your inputs.

**Comment by krwq** [comment]
@WinCPP have you build repo from the git root directory first (build.cmd)? I've seen multiple projects having that layout and they worked correctly

**Comment by WinCPP** [comment]
@krwq I did a rebase with dotnet/master, cleaned and did a .\build.cmd, just to ensure everything is built. But still the tests failed as I mentioned in previous comment i.e. by executing `msbuild /t:RebuildAndTest" in the folder `src\System.Private.Xml\tests\XPath\XPathDocument`

I found that executing `.\build.cmd src\System.Private.Xml\tests` at the root of the CoreFx repo worked and so did `.\build.cmd src\System.Private.Xml\tests\XPath\XmlDocument`. Not sure how to have `msbuild` work for individual folder. But I think I am good for now...

Thanks!

**Comment by krwq** [comment]
@WinCPP, try passing also full csproj name to msbuild, other things you can try is to use Developer Command Prompt and try msbuilding from there (looking at the paths I think you might be using Powershell but I might be wrong). Also could you share the error or full log you are seeing?

**Comment by WinCPP** [comment]
@krwq, I am using developer command prompt... I just truncated the path to keep only the relevant bits...

The entire command with complete prompt is as follows.

```
D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument>msbuild /t:RebuildAndTest System.Xml.XPath.Tests.csproj
```
The entire output is as follows. Please note that testResults.xml is not generated although the end of the log below asks us to check that... So please ignore that...

```
D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument>msbuild /t:RebuildAndTest System.Xml.XPath.Tests.csproj
Microsoft (R) Build Engine version 15.3.409.57025 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 11/2/2017 11:49:59 PM.
Project "D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj" on node 1 (RebuildAndTest target(s)).
CoreClean:
  Deleting file "D:\WinCPP\corefx\bin\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\System.Xml.XPath.Tests.dll".
  Deleting file "D:\WinCPP\corefx\bin\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\System.Xml.XPath.Tests.pdb".
  Deleting file "D:\WinCPP\corefx\bin\obj\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\System.Xml.XPath.Tests.csprojResolveAssemblyReference.cache".
  Deleting file "D:\WinCPP\corefx\bin\obj\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\System.Xml.XPath.Tests.csproj.CoreCompileInputs.cache".
  Deleting file "D:\WinCPP\corefx\bin\obj\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\_AssemblyInfo.cs".
  Deleting file "D:\WinCPP\corefx\bin\obj\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\disabledAnalyzers.config".
  Deleting file "D:\WinCPP\corefx\bin\obj\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\System.Xml.XPath.Tests.dll".
  Deleting file "D:\WinCPP\corefx\bin\obj\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\System.Xml.XPath.Tests.pdb".
CoreCompile:
  D:\WinCPP\corefx\Tools\net46\roslyn\tools\csc.exe /noconfig /nowarn:1701,1702 /nostdlib+ /warn:4 /define:DEBUG;TRACE;FEATURE_XML_XPATH_ID;DEBUGRESOURCES;SIGNED /reference:D:\Win
  CPP\corefx\bin\ref\netcoreapp\CoreFx.Private.TestUtilities.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\Microsoft.CSharp.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp
  \Microsoft.VisualBasic.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\Microsoft.Win32.Primitives.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\Microsoft.Win32.Registry.
  AccessControl.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\Microsoft.Win32.Registry.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\mscorlib.dll /reference:D:\WinCPP\co
  refx\bin\ref\netcoreapp\netstandard.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.AppContext.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Buffers.dll /r
  eference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.CodeDom.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Collections.Concurrent.dll /reference:D:\WinCPP\corefx\bin\
  ref\netcoreapp\System.Collections.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Collections.Immutable.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Colle
  ctions.NonGeneric.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Collections.Specialized.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ComponentModel.Anno
  tations.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ComponentModel.Composition.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ComponentModel.DataAnnotat
  ions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ComponentModel.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ComponentModel.EventBasedAsync.dll /refer
  ence:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ComponentModel.Primitives.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ComponentModel.TypeConverter.dll /reference:D
  :\WinCPP\corefx\bin\ref\netcoreapp\System.Composition.AttributedModel.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Composition.Convention.dll /reference:D:\WinCPP\c
  orefx\bin\ref\netcoreapp\System.Composition.Hosting.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Composition.Runtime.dll /reference:D:\WinCPP\corefx\bin\ref\netcore
  app\System.Composition.TypedParts.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Configuration.ConfigurationManager.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp
  \System.Configuration.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Console.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Core.dll /reference:D:\WinCPP\c
  orefx\bin\ref\netcoreapp\System.Data.Common.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Data.DataSetExtensions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\S
  ystem.Data.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Data.Odbc.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Data.SqlClient.dll /reference:D:\WinCPP\
  corefx\bin\ref\netcoreapp\System.Diagnostics.Contracts.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.Debug.dll /reference:D:\WinCPP\corefx\bin\ref\netcor
  eapp\System.Diagnostics.DiagnosticSource.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.EventLog.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System
  .Diagnostics.FileVersionInfo.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.PerformanceCounter.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.D
  iagnostics.Process.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.StackTrace.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.TextWri
  terTraceListener.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.Tools.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.TraceSource.dl
  l /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Diagnostics.Tracing.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.DirectoryServices.AccountManagement.dll /re
  ference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.DirectoryServices.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.DirectoryServices.Protocols.dll /reference:D:\WinC
  PP\corefx\bin\ref\netcoreapp\System.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Drawing.Common.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Drawing.dl
  l /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Drawing.Primitives.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Dynamic.Runtime.dll /reference:D:\WinCPP\cor
  efx\bin\ref\netcoreapp\System.Globalization.Calendars.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Globalization.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\
  System.Globalization.Extensions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.Compression.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.Compression
  .FileSystem.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.Compression.ZipFile.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.dll /reference:D:\WinCP
  P\corefx\bin\ref\netcoreapp\System.IO.FileSystem.AccessControl.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.FileSystem.dll /reference:D:\WinCPP\corefx\bin\ref\ne
  tcoreapp\System.IO.FileSystem.DriveInfo.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.FileSystem.Primitives.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\Sys
  tem.IO.FileSystem.Watcher.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.IsolatedStorage.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.MemoryMappedF
  iles.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.Packaging.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.Pipes.AccessControl.dll /reference:D:\Wi
  nCPP\corefx\bin\ref\netcoreapp\System.IO.Pipes.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.Ports.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.IO.Un
  managedMemoryStream.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Json.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Linq.dll /reference:D:\WinCPP\corefx
  \bin\ref\netcoreapp\System.Linq.Expressions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Linq.Parallel.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Lin
  q.Queryable.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Management.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Memory.dll /reference:D:\WinCPP\corefx
  \bin\ref\netcoreapp\System.Net.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.Http.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.Http.Rtc.dll /ref
  erence:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.Http.WinHttpHandler.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.HttpListener.dll /reference:D:\WinCPP\cor
  efx\bin\ref\netcoreapp\System.Net.Mail.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.NameResolution.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net
  .NetworkInformation.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.Ping.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.Primitives.dll /reference:D:
  \WinCPP\corefx\bin\ref\netcoreapp\System.Net.Requests.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.Security.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\S
  ystem.Net.ServicePoint.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.Sockets.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.WebClient.dll /referen
  ce:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.WebHeaderCollection.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.WebProxy.dll /reference:D:\WinCPP\corefx\bin\
  ref\netcoreapp\System.Net.WebSockets.Client.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Net.WebSockets.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Nu
  merics.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Numerics.Vectors.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ObjectModel.dll /reference:D:\WinCPP\
  corefx\bin\ref\netcoreapp\System.Reflection.Context.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Reflection.DispatchProxy.dll /reference:D:\WinCPP\corefx\bin\ref\ne
  tcoreapp\System.Reflection.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Reflection.Emit.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Reflection.Emit.IL
  Generation.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Reflection.Emit.Lightweight.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Reflection.Extensions.
  dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Reflection.Metadata.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Reflection.Primitives.dll /reference:D:\W
  inCPP\corefx\bin\ref\netcoreapp\System.Reflection.TypeExtensions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Resources.Reader.dll /reference:D:\WinCPP\corefx\bin\r
  ef\netcoreapp\System.Resources.ResourceManager.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Resources.Writer.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\Syst
  em.Runtime.CompilerServices.Unsafe.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.CompilerServices.VisualC.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\
  System.Runtime.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.Extensions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.Handles.dll /refere
  nce:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.InteropServices.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.InteropServices.RuntimeInformation.dll /
  reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.InteropServices.WindowsRuntime.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.Intrinsics.X86.dll
  /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.Loader.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.Numerics.dll /reference:D:\WinCPP\corefx\b
  in\ref\netcoreapp\System.Runtime.Serialization.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.Serialization.Formatters.dll /reference:D:\WinCPP\corefx\bin\ref
  \netcoreapp\System.Runtime.Serialization.Json.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Runtime.Serialization.Primitives.dll /reference:D:\WinCPP\corefx\bin\ref\
  netcoreapp\System.Runtime.Serialization.Xml.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.AccessControl.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\S
  ystem.Security.Claims.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Cryptography.Algorithms.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Securi
  ty.Cryptography.Cng.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Cryptography.Csp.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Crypto
  graphy.Encoding.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Cryptography.OpenSsl.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Crypto
  graphy.Pkcs.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Cryptography.Primitives.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Cryptog
  raphy.ProtectedData.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Cryptography.X509Certificates.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Se
  curity.Cryptography.Xml.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Permissions.dll /re
  ference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Principal.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Security.Principal.Windows.dll /reference:D:\WinC
  PP\corefx\bin\ref\netcoreapp\System.Security.SecureString.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ServiceModel.Syndication.dll /reference:D:\WinCPP\corefx\bin\
  ref\netcoreapp\System.ServiceModel.Web.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ServiceProcess.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Service
  Process.ServiceController.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Text.Encoding.CodePages.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Text.Encodi
  ng.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Text.Encoding.Extensions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Text.Encodings.Web.dll /reference
  :D:\WinCPP\corefx\bin\ref\netcoreapp\System.Text.RegularExpressions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.AccessControl.dll /reference:D:\WinCPP\co
  refx\bin\ref\netcoreapp\System.Threading.Channels.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.T
  hreading.Overlapped.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.Tasks.Dataflow.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.Tasks.
  dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.Tasks.Extensions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.Tasks.Parallel.dll /refe
  rence:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.Thread.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Threading.ThreadPool.dll /reference:D:\WinCPP\corefx\
  bin\ref\netcoreapp\System.Threading.Timer.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Transactions.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Transa
  ctions.Local.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.ValueTuple.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Web.dll /reference:D:\WinCPP\corefx\b
  in\ref\netcoreapp\System.Web.HttpUtility.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Windows.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Xml.dll /ref
  erence:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Xml.Linq.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Xml.ReaderWriter.dll /reference:D:\WinCPP\corefx\bin\ref\net
  coreapp\System.Xml.Serialization.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Xml.XDocument.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Xml.XmlDocumen
  t.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Xml.XmlSerializer.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\System.Xml.XPath.dll /reference:D:\WinCPP\corefx
  \bin\ref\netcoreapp\System.Xml.XPath.XDocument.dll /reference:D:\WinCPP\corefx\bin\ref\netcoreapp\WindowsBase.dll /reference:D:\WinCPP\corefx\bin\runtime\netcoreapp-Windows_NT-D
  ebug-x64\xunit.core.dll /reference:D:\WinCPP\corefx\bin\runtime\netcoreapp-Windows_NT-Debug-x64\Xunit.NetCore.Extensions.dll /reference:D:\WinCPP\corefx\bin\runtime\netcoreapp-W
  indows_NT-Debug-x64\xunit.assert.dll /reference:D:\WinCPP\corefx\bin\runtime\netcoreapp-Windows_NT-Debug-x64\xunit.abstractions.dll /reference:D:\WinCPP\corefx\bin\runtime\netco
  reapp-Windows_NT-Debug-x64\xunit.performance.core.dll /reference:D:\WinCPP\corefx\bin\runtime\netcoreapp-Windows_NT-Debug-x64\xunit.performance.api.dll /debug+ /debug:Portable /
  delaysign- /keyfile:D:\WinCPP\corefx\Tools/Test.snk /optimize- /out:D:\WinCPP\corefx\bin/obj/Windows_NT.AnyCPU.Debug/System.Xml.XPath.Tests/netcoreapp/System.Xml.XPath.Tests.dll
   /ruleset:D:\WinCPP\corefx\CodeAnalysis.ruleset /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\91893.xml,System.Xml.XPath.Tests.TestData.91893.xml /resou
  rce:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\JPN_problem_chars_1.xml,System.Xml.XPath.Tests.TestData.JPN_problem_chars_1.xml /resource:D:\WinCPP\corefx\src\C
  ommon\tests\\System\Xml\XPath\TestData\JPN_problem_chars_2.xml,System.Xml.XPath.Tests.TestData.JPN_problem_chars_2.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XP
  ath\TestData\KOR_problem_chars_b.xml,System.Xml.XPath.Tests.TestData.KOR_problem_chars_b.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\Russian_probl
  em_chars.xml,System.Xml.XPath.Tests.TestData.Russian_problem_chars.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\Single_byte_problem_chars_b.xml,Sys
  tem.Xml.XPath.Tests.TestData.Single_byte_problem_chars_b.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\Surrogates_1.xml,System.Xml.XPath.Tests.TestD
  ata.Surrogates_1.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\Thai_risky_chars.xml,System.Xml.XPath.Tests.TestData.Thai_risky_chars.xml /resource:D
  :\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\XQL_Orders_j1.xml,System.Xml.XPath.Tests.TestData.XQL_Orders_j1.xml /resource:D:\WinCPP\corefx\src\Common\tests\\Syst
  em\Xml\XPath\TestData\XQL_Orders_j3.xml,System.Xml.XPath.Tests.TestData.XQL_Orders_j3.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xql_orders-flat-
  200a.xml,System.Xml.XPath.Tests.TestData.xql_orders-flat-200a.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\books.xml,System.Xml.XPath.Tests.TestDat
  a.books.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\books_2.xml,System.Xml.XPath.Tests.TestData.books_2.xml /resource:D:\WinCPP\corefx\src\Common\
  tests\\System\Xml\XPath\TestData\bookstore.xml,System.Xml.XPath.Tests.TestData.bookstore.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\data.xml,Syst
  em.Xml.XPath.Tests.TestData.data.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\data1.xml,System.Xml.XPath.Tests.TestData.data1.xml /resource:D:\WinC
  PP\corefx\src\Common\tests\\System\Xml\XPath\TestData\data4.xml,System.Xml.XPath.Tests.TestData.data4.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\
  dummy.xml,System.Xml.XPath.Tests.TestData.dummy.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\id4.xml,System.Xml.XPath.Tests.TestData.id4.xml /resou
  rce:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\lang.xml,System.Xml.XPath.Tests.TestData.lang.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\
  TestData\name.xml,System.Xml.XPath.Tests.TestData.name.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\name2.xml,System.Xml.XPath.Tests.TestData.name2
  .xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\name4.xml,System.Xml.XPath.Tests.TestData.name4.xml /resource:D:\WinCPP\corefx\src\Common\tests\\Syst
  em\Xml\XPath\TestData\namespaces.xml,System.Xml.XPath.Tests.TestData.namespaces.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\ns_default.xml,System.
  Xml.XPath.Tests.TestData.ns_default.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\ns_prefixes.xml,System.Xml.XPath.Tests.TestData.ns_prefixes.xml /r
  esource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\numbers.xml,System.Xml.XPath.Tests.TestData.numbers.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\
  Xml\XPath\TestData\space.xml,System.Xml.XPath.Tests.TestData.space.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\t114730.xml,System.Xml.XPath.Tests.
  TestData.t114730.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\t98598.xml,System.Xml.XPath.Tests.TestData.t98598.xml /resource:D:\WinCPP\corefx\src\
  Common\tests\\System\Xml\XPath\TestData\test1.xml,System.Xml.XPath.Tests.TestData.test1.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\test63682.xml,
  System.Xml.XPath.Tests.TestData.test63682.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\test63733.xml,System.Xml.XPath.Tests.TestData.test63733.xml
  /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\test66246.xml,System.Xml.XPath.Tests.TestData.test66246.xml /resource:D:\WinCPP\corefx\src\Common\tests\\S
  ystem\Xml\XPath\TestData\turkish.xml,System.Xml.XPath.Tests.TestData.turkish.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xp001.xml,System.Xml.XPat
  h.Tests.TestData.xp001.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xp002.xml,System.Xml.XPath.Tests.TestData.xp002.xml /resource:D:\WinCPP\corefx\
  src\Common\tests\\System\Xml\XPath\TestData\xp003.xml,System.Xml.XPath.Tests.TestData.xp003.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xp004.xml,
  System.Xml.XPath.Tests.TestData.xp004.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xp005.xml,System.Xml.XPath.Tests.TestData.xp005.xml /resource:D:
  \WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xp006.xml,System.Xml.XPath.Tests.TestData.xp006.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\Test
  Data\xp008.xml,System.Xml.XPath.Tests.TestData.xp008.xml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xp012.xml,System.Xml.XPath.Tests.TestData.xp012.x
  ml /resource:D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestData\xpC001.xml,System.Xml.XPath.Tests.TestData.xpC001.xml /target:library /warnaserror+ /utf8output /source
  link:D:\WinCPP\corefx\bin/obj/source_link.json /langversion:latest /analyzer:D:\WinCPP\corefx\Tools/net46/analyzers/Microsoft.DotNet.CodeAnalysis.dll /additionalfile:D:\WinCPP\c
  orefx\Tools\PinvokeAnalyzer_Win32Apis.txt /additionalfile:D:\WinCPP\corefx\bin/obj/Windows_NT.AnyCPU.Debug/System.Xml.XPath.Tests/netcoreapp//disabledAnalyzers.config CreateNavi
  gatorFromXmlReader.cs XmlReaderXPathTest.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\Common\FileHelper.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\Common\I
  CreateNavigator.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\Common\Utils.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\Common\XPathResult.cs D:\WinCPP\corefx
  \src\Common\tests\\System\Xml\XPath\Common\XPathResultToken.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\CoreFunctionLibrary\BooleanFunctionsTests.cs D:\WinCPP\corefx\
  src\Common\tests\\System\Xml\XPath\CoreFunctionLibrary\ComplexExpressionsTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\CoreFunctionLibrary\NodeSetFunctionsTests.c
  s D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\CoreFunctionLibrary\NodeSetMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\CoreFunctionLibrary\NumberFu
  nctionsTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\CoreFunctionLibrary\ParameterTypeCoercionTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\CoreFun
  ctionLibrary\StringFunctionsTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\CustomerScenarios\CustomerScenariosTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xm
  l\XPath\CustomerScenarios\MatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncExpressions\BasicMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\
  XPath\FuncExpressions\BasicsTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncExpressions\BooleansTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\Fun
  cExpressions\NodeSetsMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncExpressions\NodeSetsTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\Fun
  cExpressions\NumbersTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncExpressions\SetContextFunctionalTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath
  \FuncExpressions\SetContextMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathAbbreviatedSyntaxMatchesTests.cs D:\WinCPP\corefx\src\Common\test
  s\\System\Xml\XPath\FuncLocation\PathAbbreviatedSyntaxTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathAxeCombinationsTests.cs D:\WinCPP\corefx\src\
  Common\tests\\System\Xml\XPath\FuncLocation\PathAxeComplexExprMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathAxeComplexExprTests.cs D:\WinC
  PP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathAxeMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathAxesTests.cs D:\WinCPP\core
  fx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathNodeMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathNodeTests.cs D:\WinCPP\corefx\src
  \Common\tests\\System\Xml\XPath\FuncLocation\PathPredicateComplexExprTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathPredicateMatchesTests.cs D:\Wi
  nCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathPredicatePositionTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\PathPredicatesTests.cs
   D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\StepMatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\FuncLocation\StepsTests.cs D:\WinCPP\co
  refx\src\Common\tests\\System\Xml\XPath\MiscellaneousCases\GlobalizationTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\MiscellaneousCases\LexicalStructureTests.cs
  D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\MiscellaneousCases\MatchesTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\MiscellaneousCases\MiscellaneousCasesT
  ests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\MiscellaneousCases\NodeSetsTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\MiscellaneousCases\Regression
  TestsTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\TestsWithEncoding\MiscWithEncodings.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\XPathExpressionTests
  \CompileTests.cs D:\WinCPP\corefx\src\Common\tests\\System\Xml\XPath\XPathExpressionTests\EvaluateTests.cs D:\WinCPP\corefx\bin/obj/Windows_NT.AnyCPU.Debug/System.Xml.XPath.Test
  s/netcoreapp/_AssemblyInfo.cs @D:\WinCPP\corefx\Tools\checksum.rsp
  Using shared compilation with compiler from directory: D:\WinCPP\corefx\Tools\net46\roslyn\tools
CopyFilesToOutputDirectory:
  Copying file from "D:\WinCPP\corefx\bin/obj/Windows_NT.AnyCPU.Debug/System.Xml.XPath.Tests/netcoreapp/System.Xml.XPath.Tests.dll" to "D:\WinCPP\corefx\bin/Windows_NT.AnyCPU.Debu
  g/System.Xml.XPath.Tests/netcoreapp/System.Xml.XPath.Tests.dll".
  System.Xml.XPath.Tests -> D:\WinCPP\corefx\bin\Windows_NT.AnyCPU.Debug\System.Xml.XPath.Tests\netcoreapp\System.Xml.XPath.Tests.dll
  Copying file from "D:\WinCPP\corefx\bin/obj/Windows_NT.AnyCPU.Debug/System.Xml.XPath.Tests/netcoreapp/System.Xml.XPath.Tests.pdb" to "D:\WinCPP\corefx\bin/Windows_NT.AnyCPU.Debu
  g/System.Xml.XPath.Tests/netcoreapp/System.Xml.XPath.Tests.pdb".
GenerateTestExecutionScripts:
  Generating D:\WinCPP\corefx\bin//TestDependencies/System.Xml.XPath.Tests-.dependencylist.txt
  Test Command lines = call %RUNTIME_PATH%\dotnet.exe xunit.console.netcore.exe System.Xml.XPath.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetco
  reapptests -notrait category=nonwindowstests  -notrait category=OuterLoop -notrait category=failing
  Wrote Windows-compatible test execution script to D:\WinCPP\corefx\bin/tests/System.Xml.XPath.Tests/netcoreapp-Windows_NT-Debug-x64//RunTests.cmd
RunTestsForProject:
  D:\WinCPP\corefx\bin/tests/System.Xml.XPath.Tests/netcoreapp-Windows_NT-Debug-x64//RunTests.cmd D:\WinCPP\corefx\bin/testhost/netcoreapp-Windows_NT-Debug-x64/
  Using D:\WinCPP\corefx\bin\testhost\netcoreapp-Windows_NT-Debug-x64\ as the test runtime folder.
  Executing in D:\WinCPP\corefx\bin\tests\System.Xml.XPath.Tests\netcoreapp-Windows_NT-Debug-x64\
  Running tests... Start time: 23:50:03.28
  xUnit.net console test runner (64-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.

  error: file not found: System.Xml.XPath.Tests.dll
  Finished running tests.  End time=23:50:05.53, Exit code = 3
D:\WinCPP\corefx\Tools\tests.targets(484,5): warning MSB3073: The command "D:\WinCPP\corefx\bin/tests/System.Xml.XPath.Tests/netcoreapp-Windows_NT-Debug-x64//RunTests.cmd D:\WinCP
P\corefx\bin/testhost/netcoreapp-Windows_NT-Debug-x64/" exited with code 3. [D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj]
  The previous error was converted to a warning because the task was called with ContinueOnError=true.
  Build continuing because "ContinueOnError" on the task "Exec" is set to "true".
D:\WinCPP\corefx\Tools\tests.targets(492,5): error : One or more tests failed while running tests from 'System.Xml.XPath.Tests' please check D:\WinCPP\corefx\bin/tests/System.Xml.
XPath.Tests/netcoreapp-Windows_NT-Debug-x64/testResults.xml for details! [D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj]
Done Building Project "D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj" (RebuildAndTest target(s)) -- FAILED.


Build FAILED.

"D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj" (RebuildAndTest target) (1) ->
(RunTestsForProject target) ->
  D:\WinCPP\corefx\Tools\tests.targets(484,5): warning MSB3073: The command "D:\WinCPP\corefx\bin/tests/System.Xml.XPath.Tests/netcoreapp-Windows_NT-Debug-x64//RunTests.cmd D:\Win
CPP\corefx\bin/testhost/netcoreapp-Windows_NT-Debug-x64/" exited with code 3. [D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj]


"D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj" (RebuildAndTest target) (1) ->
(RunTestsForProject target) ->
  D:\WinCPP\corefx\Tools\tests.targets(492,5): error : One or more tests failed while running tests from 'System.Xml.XPath.Tests' please check D:\WinCPP\corefx\bin/tests/System.Xm
l.XPath.Tests/netcoreapp-Windows_NT-Debug-x64/testResults.xml for details! [D:\WinCPP\corefx\src\System.Private.Xml\tests\XPath\XPathDocument\System.Xml.XPath.Tests.csproj]

    1 Warning(s)
    1 Error(s)

Time Elapsed 00:00:05.98
```

**Comment by WinCPP** [comment]
Between, since `msbuild` from the \test\XPath\XPathNavigator (last one is new folder, not yet committed) gives above mentioned error, I am not sure how to generate coverage report. I tried `build-tests -coverage` (as per the documentation) but that turned out to be full run; too much of time; so cancelled it. I am looking for some way to run coverage report just for the .\src\System.Private.Xml folder. Appreciate inputs on the same. Thanks!

**Comment by krwq** [comment]
@WinCPP I was able to repro - I'll send PR shortly with the fix - currently testing locally (see the issue for workaround)

**Comment by krwq** [comment]
@WinCPP if you rebase it should work now - please let me know if you encounter any problems

**Comment by WinCPP** [comment]
Thanks @krwq! msbuild works now and I am able to generate coverage report as well.

**Comment by WinCPP** [comment]
@danmosemsft @krwq While importing tests from the mono project (discussion above), I have encountered a problem with regards to `XPathNavigatorReader.LocalName` (please refer to the code [here](https://github.com/dotnet/corefx/blob/63e70fa6b1bc6b7175467e0ad16cbbf4c44f8d83/src/System.Private.Xml/src/System/Xml/XPath/XPathNavigatorReader.cs#L213-L225)).

I am not sure if I am seeing an anomaly with regards to the implementation of the mentioned property. Kindly let me know what is the correct channel to seek inputs on the code or, should I directly open an issue in github, regardless of whether it is indeed a defect or gap in my understanding...?

Thanks!

**Comment by krwq** [comment]
@WinCPP, if you believe there is something wrong with the implementation please create a new issue and write the details what is expected and what you get (or point to a failing test case) and we can have conversation over there if there are any doubts

**Comment by WinCPP** [comment]
@krwq I have created #25133 for the confusing behavior that I mentioned above...
